pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1011,4 +1012,33 @@ public void testStatementDescribe() throws SQLException {
     pstmt.close();
   }
 
+  @Test
+  public void testSetTimeAndTimestamp() throws SQLException {
+    PreparedStatement ps = con.prepareStatement("SELECT ?::timestamp");
+    try {
+      Timestamp ts = new Timestamp(1474997614836L);
+      for (int i = 0; i < 3; ++i) {
+        ResultSet rs;
+        ps.setNull(1, Types.DATE);
+        rs = ps.executeQuery();
+        try {
+          assertTrue(rs.next());
+          assertNull(rs.getObject(1));
```

Comment:
>a lot of code that uses assertTrue for rs.next() and assertEquals without a message.

That is unfortunate. New code should explain what it is trying to test.

The thing is when test fails, it takes lots of time to figure out what was the intention behind the test and why did it fail.
Having decent messages greatly reduces maintenance costs.